File: codeforces/19.queue_at_the_school/qats.cpp
#include <iostream>
#include <cstring>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int n, t;
  string s;
  cin >> n >> t;
  cin >> s;
  while (t--) {
    for (unsigned short int i = 0; i < (n - 1); i++) {
      if (s[i] == 'B' && s[i + 1] == 'G') {
        s[i] = 'G';
        s[i + 1] = 'B';
        i++;
      }
    }
  }
  cout << s << endl;
  return 0;
}

File: codeforces/low_difficulty/800_points/19.queue_at_the_school_[266B]/qats.py
# Author: José Rodolfo (jric2002)
input_data = input().split(" ")
n = int(input_data[0])
t = int(input_data[1])
s = list(input())
while (t):
  i = 0
  while (i < (n - 1)):
    if (s[i] == 'B' and s[i + 1] == 'G'):
      s[i] = 'G'
      s[i + 1] = 'B'
      i += 1
    i += 1
  t -= 1
print("".join(s))